atomicapp/nulecule/config.py

```diff
@@ -0,0 +1,176 @@
+import copy
+import logging
+
+from atomicapp.constants import (GLOBAL_CONF,
+                                 LOGGER_COCKPIT,
+                                 DEFAULT_PROVIDER,
+                                 NAMESPACE_SEPARATOR)
+from collections import defaultdict
+
+cockpit_logger = logging.getLogger(LOGGER_COCKPIT)
+
+
+class Config(object):
+    """
+    Store config data for a Nulecule or Nulecule component.
+
+    It stores config data from different sources (answers, cli, user data)
+    separately, and exposes high level interfaces to read, write data
+    from it, with enforced read/write policies.
+    """
+
+    def __init__(self, namespace='', answers=None, cli=None,
+                 data=None, is_nulecule=False):
+        self._namespace = namespace
+        self._is_nulecule = is_nulecule or False
+        self._parent_ns, self._current_ns = self._split_namespace(self._namespace)
+        # Store answers data
+        self._answers = defaultdict(dict)
+        self._answers.update(answers or {})
+        # Store CLI data
+        self._cli = cli or {}
+        # Store data collected during runtime
+        self._data = data or defaultdict(dict)
+        self._context = None
+        self._provider = None
```

Comment:
These are used in init anyways, they won't be able to access them unless you create it as a python read/write attribute